dagshub/streaming/filesystem.py

```diff
@@ -282,12 +293,44 @@ def open(self, file: Union[bytes, PathLike, int], mode: str = 'r', *args, opener
                             _ = self.stat(self.project_root / relative_path.parent)
                         except FileNotFoundError:
                             raise err
-                        return self.__open(relative_path, mode, opener=project_root_opener)
+                        # Try to download the file if we're in append modes
+                        if "a" in mode or "+" in mode:
+                            resp = self._api_download_file_git(relative_path)
+                            if resp.ok:
+                                with self.__open(relative_path, 'wb', opener=project_root_opener) as output:
+                                    output.write(resp.content)
+                        return self.__open(relative_path, mode, buffering, encoding, errors, newline,
+                                           closefd, opener=project_root_opener)
 
         else:
-            return self.__open(file, mode, *args, **kwargs, opener=opener)
+            return self.__open(file, mode, buffering, encoding, errors, newline,
+                               closefd, opener)
 
-    def stat(self, path: Union[str, bytes, PathLike], *, dir_fd=None, follow_symlinks=True):
+    def os_open(self, path, flags, mode=0o777, *, dir_fd=None):
+        """
+        os.open is supposed to be lower level, but it's still being used by e.g. Pathlib
+        We're trying to wrap around it here, by parsing flags and calling the higher-level open
+        Caveats: list of flags being handled is not exhaustive + mode doesn't work
+                 (because we lose them when passing to builtin open)
+        WARNING: DO NOT patch actual os.open, because the builtin uses os.open.
+                 This is only for the purposes of patching pathlib.open
+        """
+        if dir_fd is not None:
+            logger.debug("fs.os_open - NotImplemented")
+            raise NotImplementedError('DagsHub\'s patched os.open() (for pathlib only) does not support dir_fd')
+        try:
+            open_mode = "r"
+            # Write modes - calling in append mode, so we create the folders if file doesn't exist, but not truncate it
```

Comment:
Should be better explained like you explained to me verbally - to create any intermediate folders IFF the file exists in the repo